db/migrate/20190509142148_update_custom_button_sets.rb

```diff
@@ -0,0 +1,21 @@
+class UpdateCustomButtonSets < ActiveRecord::Migration[5.2]
+  class MiqSet < ActiveRecord::Base
+    serialize :set_data
+  end
+
+  class CustomButton < ActiveRecord::Base; end
+
+  def up
+    say_with_time("Removing deleted buttons from Custom Button Sets") do
+      MiqSet.select(:id, :set_data).where(:set_type => 'CustomButtonSet').each do |cbs|
+        next if cbs.set_data[:button_order].blank?
+        existing_buttons = CustomButton
+          .where(:id => cbs.set_data[:button_order])
+          .order("position(id::text in '#{cbs.set_data[:button_order].join(',')}')")
```

Comment:
Ugh, that looks like an SQL injection.
We can *believe* `button_order` only contains ints, but unless we check it, this is dangerous.

Reply:
@lpichler as you've corrected me not to use Ruby sort, is there a good reason not to use the Ruby sort?

Reply:
I am not against ruby solution I knew that we are using implementation in SQL

https://github.com/ManageIQ/manageiq/pull/18060/files#diff-72d768ae2e73b9e77080d4a7df59cf56R4

so you can copy `with_array_order` here or use ruby.